lib/Jsrt/Jsrt.cpp

```diff
@@ -4460,19 +4462,40 @@ CHAKRA_API JsSerialize(
         scriptFlag = LoadScriptFlag_None;
     }
 
-    return JsSerializeScriptCore(script, cb, scriptFlag, nullptr,
-        0, buffer, bufferSize, scriptVal);
+    unsigned int bufferSize = 0;
+    JsErrorCode errorCode = JsSerializeScriptCore(script, cb, scriptFlag, nullptr,
+        0, nullptr, &bufferSize, scriptVal);
+
+    if (errorCode != JsNoError)
+    {
+        return errorCode;
+    }
+
+    if (bufferSize == 0)
+    {
+        return JsErrorScriptCompile;
+    }
+
+    if ((errorCode = JsCreateArrayBuffer(bufferSize, bufferVal)) == JsNoError)
```

Comment:
It's fine either way. It's worth noting that in other places, such as above, the assignment to error code is done before the condition that checks it. Also separating the two doesn't hurt readability either.